### PR TITLE
Initial copy of the use testing library adr

### PR DESCRIPTION
**Pull request by YanZ777**
Copy over from Google Drive and adjust some formatting

### CODE REVIEW

File: practices/appeng/adrs/0004-use-testing-library.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+* `+` Automated tests require a leap of faith, while manual testing feels more tangible.
+* `+` Can test more edge cases and states with manual testing that may not have been found or written with automated testing
+* `-` Extra time is spent doing manual testing
+* `-` Manual testing is slow and doesn’t scale
+* `-` Much tighter feedback loops with automated testing, which can test a wide range of regressions much more quickly than manual testing.
+* `-` Between Jest and Cypress, which Testing Library works with, it’s very difficult to think of a test scenario that can be manually, but not programmatically, tested.  If it’s important enough to test, very likely it can be automated.
+
+## Circumstances to not use this ADR / Exceptions
+
+There will be projects where existing code has tests already written in other styles, not using Testing Library.  In those cases, the team will have to decide whether the effort to convert is worthwhile.  If something like Enzyme is blocking upgrading other libraries, it may be worthwhile, but in some cases the team might decide that the conversion time is better spent elsewhere.
```
Comment (jenniferlynparsons) [inline practices/appeng/adrs/0004-use-testing-library.md:71]: QUESTION: `something like Enzyme` is this meant to refer to Enzyme specifically or any framework?

**Review by jenniferlynparsons — APPROVED** [review]
Just dropped one small non-blocking nitpick in there but otherwise this looks good!

**Comment by rpdelaney** [comment]
I tried merging this, but the PR is older than the docusaurus migration so the merge conflict is pretty nasty. If we still need it, I'd recommend recreating in a new PR. If you disagree, please feel free to reopen :)
